### ISSUE TITLE
Starting linter on a remote resource will block indefinitely 

### ISSUE THREAD

**Issue by maximveksler** [sindresorhus/awesome-lint#53]
The following never stops 

```
awesome-lint 'https://gist.githubusercontent.com/maximveksler/e04db4c4e50b5756f8865a836028bacf/raw/6fdc20ef697e2e2445e1c52a14012c1f32797344/README.md'
⠙ Linting
```

**Comment by sindresorhus** [comment]
It should be the URL to a repo or no argument. We should throw an error on non-repo arguments then.

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $20.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/64659785/issues/53) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/64659785/issues/53).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/users/44827199/repositories) on IssueHunt to raise funds.

**Comment by chinesedfan** [comment]
Checking the command `git clone` status costs too much time(git timeout) and the package `git-clone` doesn't support to throw out error immediately. How about simply checking whether the url is end with `.git`?

**Comment by sindresorhus** [comment]
> How about simply checking whether the url is end with .git?

The `.git` part is optional when cloning repos from GitHub.

**Comment by sindresorhus** [comment]
However, since we only support GitHub repos, we can just validate that the URL is valid repo URL.

**Comment by itaisteinherz** [comment]
This seems related to #68.

**Comment by IssuehuntBot** [comment]
@sindresorhus has rewarded $18.00 to @chinesedfan. [See it on IssueHunt](https://issuehunt.io/repos/64659785/issues/53)

- :moneybag: Total deposit: $20.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $2.00
